examples/client_simple.rs

```diff
@@ -55,5 +55,6 @@ pub fn main() {
         State::Finished(MessageSent::No) => {
             panic!("PLAIN exchange produced no output")
         }
+        _ => unimplemented!(),
```

Comment:
This should be documented

Reply:
It's IMO safe to make `State` exhaustive i.e. remove the `#[non_exhaustive]` from it.

It does comprehensively cover all bases, as there is no way in SASL for a mechanism to keep running but not have sent at least an empty message.